src/filter/median.rs

```diff
@@ -323,15 +323,8 @@ struct HistSet {
 
 impl HistSet {
     fn new(num_channels: u8, expected_count: u32) -> HistSet {
-        // Can't use vec![[0u32; 256], num_channels as usize]
-        // because arrays of length > 32 aren't cloneable.
-        let mut data = Vec::with_capacity(num_channels as usize);
-        for _ in 0..num_channels {
-            data.push([0u32; 256]);
-        }
-
         HistSet {
-            data,
+            data: vec![[0u32; 256]; num_channels.into()],
```

Comment:
It would be slightly nicer to merge this change and the change to stats.rs as standalone PRs rather than as part of this one. But this is only a very mild preference - I'm happy to merge them here.

Reply:
True, it was more laziness on my part as I was mid-PR when I found the change. I'll make a mini-PR for it.